scripts/e2e-test.sh

```diff
@@ -3,13 +3,25 @@ set -e # bail on errors
 
 SRC_PATH=$1
 CUSTOM_COMMAND="${2:-yarn test}"
-GATSBY_PATH="${CIRCLE_WORKING_DIRECTORY:-../../}"
+eval GATSBY_PATH=${CIRCLE_WORKING_DIRECTORY:-../..}
+TMP_LOCATION=$(mktemp -d);
+
+mkdir -p $TMP_LOCATION/$SRC_PATH
+TMP_TEST_LOCATION=$TMP_LOCATION/$SRC_PATH
+
+mkdir -p $TMP_LOCATION/scripts/
+mkdir -p $TMP_TEST_LOCATION
 
 # cypress docker does not support sudo and does not need it, but the default node executor does
-command -v gatsby-dev || command -v sudo && sudo npm install -g gatsby-dev-cli@next || npm install -g gatsby-dev-cli@next
+command -v gatsby-dev || (command -v sudo && sudo npm install -g gatsby-dev-cli@next) || npm install -g gatsby-dev-cli@next
+
+echo "Copy $SRC_PATH into $TMP_LOCATION to isolate test"
+cp -Rv $SRC_PATH/. $TMP_TEST_LOCATION
+cp -Rv $GATSBY_PATH/scripts/. $TMP_LOCATION/scripts/
 
 # setting up child integration test link to gatsby packages
-cd "$SRC_PATH"
+cd "$TMP_TEST_LOCATION"
+
 gatsby-dev --set-path-to-repo "$GATSBY_PATH"
 gatsby-dev --force-install --scan-once  # install _all_ files in gatsby/packages
```

Comment:
Is this what we want? Should we be restricting it to just the ones in package.json?

Reply:
I think the comment is false, --force-install means we don't copy but we only install through npm